src/main/java/net/starschema/clouddb/jdbc/BQSupportFuncts.java

```diff
@@ -364,14 +365,19 @@ public static String logAndGetQueryState(Job pollJob) throws IOException {
         || pollJob.getStatistics().isEmpty()) {
       throw new IOException("Failed to fetch query state.");
     }
-    BQSupportFuncts.logger.info(
-        "Job status: "
-            + pollJob.getStatus().getState()
-            + " ; "
-            + pollJob.getJobReference().getJobId()
-            + " ; "
-            + (System.currentTimeMillis() - pollJob.getStatistics().getStartTime()));
-    return pollJob.getStatus().getState();
+
+    if (Optional.ofNullable(pollJob.getStatistics().getCreationTime()).isPresent()) {
+      BQSupportFuncts.logger.info(
+          "Job status: "
+              + pollJob.getStatus().getState()
+              + " ; "
+              + pollJob.getJobReference().getJobId()
+              + " ; "
+              + (System.currentTimeMillis() - pollJob.getStatistics().getCreationTime()));
+      return pollJob.getStatus().getState();
+    } else {
+      throw new IOException("Failed to fetch creation time.");
```

Comment:
can this ever happen?
If not make it IllegalStateException or just set a default value for the nullable like 1970.

Reply:
We should not have a CreationTime empty; nor should the JobReference be null as well since it would indicate there is a job created in BQ with incomplete information or the `jobs.get` API returned a job with missing info.  Neither situation is desirable.